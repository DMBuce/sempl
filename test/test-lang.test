#!/usr/bin/env sempl
# test sempl language features
SEMPL_BRACES={{}}
#!end sempl env

# many language features are tested by the readme,
# so i haven't bothered including them here yet
#
# TODO: parameters, frontmatter, continuation, line interpolation, etc.

Make sure non-shell-identifiers aren't expanded
{$invalid.identifier}: =>{{$invalid.identifier}}<=

Check strictness of stdin parameter
{-invalid-stdin}: =>{{-invalid-stdin}}<=

Make sure field read parameter only expands other parameters
{<invalid parameter}: =>{{<invalid parameter}}<=

Check strictness of field cut parameter
{::2}: =>{{::2}}<=

Check missing variable
{$unset_variable}: =>{{$unset_variable}}<=
  Note: error message will appear at beginning of test output

Check missing files
{/nonexistent/file}: =>{{/nonexistent/file}}<=
{./nonexistent/file}: =>{{./nonexistent/file}}<=

Check missing command
{!command-not-found}: =>{{!command-not-found}}<=

