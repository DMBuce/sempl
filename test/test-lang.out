Unset environment variable: $unset_variable

# many language features are tested by the readme,
# so i haven't bothered including them here yet
#
# TODO:
# parameters
# frontmatter
# parameters
# line interpolation

Make sure non-shell-identifiers aren't expanded
{$invalid.identifier}: =>{{$invalid.identifier}}<=

Check strictness of stdin parameter
{-invalid-stdin}: =>{{-invalid-stdin}}<=

Make sure field read parameter only expands other parameters
{<invalid parameter}: =>{{<invalid parameter}}<=

Check strictness of field cut parameter
{::2}: =>{{::2}}<=

Check missing variable
{$unset_variable}: =><=
  Note: error message will appear at beginning of test output

Check missing files
cat: /nonexistent/file: No such file or directory
cat: ./nonexistent/file: No such file or directory

Check missing command
bash: line 38: command-not-found: command not found

